src/mds/MDLog.cc

```diff
@@ -822,46 +820,60 @@ void MDLog::_trim_expired_segments()
 {
   ceph_assert(ceph_mutex_is_locked_by_me(submit_mutex));
 
-  uint64_t oft_committed_seq = mds->mdcache->open_file_table.get_committed_log_seq();
+  uint64_t const oft_committed_seq = mds->mdcache->open_file_table.get_committed_log_seq();
 
   // trim expired segments?
   bool trimmed = false;
-  while (!segments.empty()) {
-    LogSegment *ls = segments.begin()->second;
+  uint64_t end = 0;
```

Comment:
@batrick I'm seeing certain failed jobs in my run which has this change. Many failures resemble the MDS unable to transition from `up:stopping` to `down:stopped`.

https://pulpito.ceph.com/vshankar-2023-03-02_09:21:58-fs-wip-vshankar-testing-20230222.044949-testing-default-smithi/7191595/

It seems like the MDS indefinitely waits for a log segment to expire (LogSegment(18/0x405552 events=1) in the test run above) or maybe something related to this log segment that does not allow the MDS to move to the next (`stopped`) state.

The list of changes in this test branch are: https://github.com/ceph/ceph/pulls?q=is%3Apr+base%3Amain+is%3Aopen+label%3Acephfs+label%3Awip-vshankar-testing1

I think this change _might_ be causing the issue.

Reply:
I've fixed this locally. Will upload changes tomorrow.